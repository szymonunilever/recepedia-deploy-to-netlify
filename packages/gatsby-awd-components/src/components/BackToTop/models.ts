import { ReactNode } from 'react';
import { ButtonViewType } from '../Button';
import { UnileverLibraryComponent } from '../../models';

export interface BackToTopProps
  extends UnileverLibraryComponent<AppContent.BackToTopContent> {
  Icon?: JSX.Element;
  children?: ReactNode | ReactNode[];
  viewType?: ButtonViewType;
  hideTopPositionPx?: number;
}

export interface CustomEventTarget {
  scrollingElement: HTMLElement | null;
}
