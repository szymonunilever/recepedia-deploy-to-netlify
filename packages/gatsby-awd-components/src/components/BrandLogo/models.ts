export interface BrandLogoProps {
  className?: string;
  brand?: string;
  linkTo?: string;
}

