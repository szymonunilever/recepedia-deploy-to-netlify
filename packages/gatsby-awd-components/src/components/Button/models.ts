import { ReactNode } from 'react';
import { Icon } from '../../models';

export interface ButtonCallback {
  // @ts-ignore
  (...arg): void;
}
export enum ButtonViewType {
  classic,
  icon,
}

export interface ButtonProps {
  className?: string;
  Icon?: Icon;
  IconSelected?: Icon;
  content?: AppContent.CTAContent;
  role?: string;
  isDisabled?: boolean;
  isSelected?: boolean;
  onClick?: ButtonCallback;
  onKeyDown?: (e: React.KeyboardEvent<HTMLElement>) => void;
  children?: ReactNode | ReactNode[];
  viewType?: ButtonViewType;
  hidden?: boolean;
  isToggle?: boolean;
  type?: 'button' | 'submit' | 'reset';
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  attributes?: Record<string, any>;
}
