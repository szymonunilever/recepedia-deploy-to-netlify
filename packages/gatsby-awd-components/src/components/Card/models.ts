import { ReactChild, ReactElement } from 'react';
import { Icon, RatingAndReviewsProvider, UnileverLibraryComponent } from '../../models';
import { ButtonProps } from '../Button';
import { RatingProps } from '../Rating';

export interface CardProps {
  key: string;
  content: CardContentProps;
  idPropertyName: string;
  ratingWidget?: ReactElement<RatingProps>
  imageSizes: string;
  className?: string;
  brand?: ReactChild;
  children?: ReactElement<ButtonProps> | ReactElement<ButtonProps>[];
}

export interface CardContentProps {
  fields:{
    slug: string
  };
  name?: string;
  title?: string;
  localImage?: Internal.LocalImage;
  images?: Internal.LocalImage[];
  averageRating?: number;
  [key: string]: any;
}

interface CardWrapper {
  key: string;
  children: ReactElement<CardProps>;
  ratingProvider: RatingAndReviewsProvider;
}

export interface RecipeCardWrapperProps extends CardWrapper{}

export interface ProductCardWrapperProps extends CardWrapper{
  mainImageIndex?: number;
}
