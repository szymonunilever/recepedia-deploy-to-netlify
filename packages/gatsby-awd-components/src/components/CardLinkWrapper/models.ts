import { ReactElement } from 'react';
import { CardProps, ProductCardWrapperProps, RecipeCardWrapperProps } from '../Card';
import { RecipeCardProps } from '../RecipeCard';

export interface CardLinkWrapperProps {
  key: string;
  isExternal?: boolean;
  title: string;
  slug: string;
  target?: string;
  href?: string;
  rel?: string;
  to?: string;
  'aria-label'?: string;
  className?: string;
  children: ReactElement<RecipeCardProps|CardProps|RecipeCardWrapperProps|ProductCardWrapperProps>;
}
