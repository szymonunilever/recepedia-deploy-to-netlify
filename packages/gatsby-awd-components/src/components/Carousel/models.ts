import { ReactChild, ReactElement } from 'react';
import { Icon } from '../../models';
import { RecipeCardProps } from '../RecipeListing/partials/models';

export interface CreateElement {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  (itemData: any): ReactElement<RecipeCardProps> | undefined;
}

export interface BreakpointProps {
  width: number;
  switchElementsBelowBreakpoint: number;
  switchElementsAfterBreakpoint: number;
  visibleElementsBelowBreakpoint: number;
  visibleElementsAboveBreakpoint: number;
}

export interface CarouselConfig {
  breakpoints?: BreakpointProps[];
  arrowIcon?: Icon;
}

export interface CarouselProps {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  list: any;
  createElementFunction: CreateElement;
  config: CarouselConfig;
  onVisibleElementsChanged?: Function;
}
