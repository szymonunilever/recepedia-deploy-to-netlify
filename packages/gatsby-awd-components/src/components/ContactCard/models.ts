import {
  titleLevel,
  UnileverLibraryComponent,
} from '../../models/globalModels';

export interface ContactCardProps extends UnileverLibraryComponent<AppContent.ContactCard> {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  Icon?: any;
  titleLevel: titleLevel;
}
