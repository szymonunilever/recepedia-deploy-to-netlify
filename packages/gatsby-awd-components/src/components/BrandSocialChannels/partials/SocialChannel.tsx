import React from 'react';
import { SocialChannelProps } from './models';
import cx from 'classnames';
import theme from './SocialChannel.module.scss';

const SocialChannel = ({
  label,
  url,
  icon,
  attributes = {},
}: SocialChannelProps) => {
  const text = label ? (
    <div className={cx(theme.socialChannels__text, 'social-channels__text')}>
      {label}
    </div>
  ) : null;

  const test = cx(theme.socialChannels__item, 'social-channels__item');
  return (
    <li className={test}>
      <a
        className={cx(theme.socialChannels__link, 'social-channels__link')}
        href={url}
        target="_blank"
        rel="noopener noreferrer nofollow"
        {...attributes}
      >
        <div
          className={cx(theme.socialChannels__icon, 'social-channels__icon')}
        >
          {icon}
        </div>
        {text}
      </a>
    </li>
  );
};

export default SocialChannel;
