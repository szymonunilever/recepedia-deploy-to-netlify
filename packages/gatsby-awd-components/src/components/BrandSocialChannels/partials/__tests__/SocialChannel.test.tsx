import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import SocialChannel from '../index';
import toJson from 'enzyme-to-json';
import { SocialChannelProps } from '../models';

describe('<SocialChannel />', () => {
  let wrapper: ReactWrapper;
  const props: SocialChannelProps = {
    label: 'Facebook',
    url: '/',
    icon: <div>Facebook</div>,
    attributes: {
      title: 'Facebook',
    },
  };

  beforeEach(() => {
    wrapper = mount(<SocialChannel {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
