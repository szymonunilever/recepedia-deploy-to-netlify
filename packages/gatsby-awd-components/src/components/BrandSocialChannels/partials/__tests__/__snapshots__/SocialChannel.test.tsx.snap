// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialChannel /> matches the snapshot 1`] = `
<SocialChannel
  attributes={
    Object {
      "title": "Facebook",
    }
  }
  icon={
    <div>
      Facebook
    </div>
  }
  label="Facebook"
  url="/"
>
  <li
    className="socialChannels__item social-channels__item"
  >
    <a
      className="socialChannels__link social-channels__link"
      href="/"
      rel="noopener noreferrer nofollow"
      target="_blank"
      title="Facebook"
    >
      <div
        className="socialChannels__icon social-channels__icon"
      >
        <div>
          Facebook
        </div>
      </div>
      <div
        className="socialChannels__text social-channels__text"
      >
        Facebook
      </div>
    </a>
  </li>
</SocialChannel>
`;
